02_bubble/sahidakhtar1/main_test.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMain(t *testing.T) {
+	//Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := "After sorting =  [1 2 3 5]\n"
+	actual := buf.String()
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+
+var testData = []struct {
+	in  []int
+	out []int
+}{
+	{[]int{3, 2, 1, 5}, []int{1, 2, 3, 5}},
+	{[]int{3, 2, 1, 5, -1}, []int{-1, 1, 2, 3, 5}},
+	{[]int{3, 2, 1, 5, -1, 3}, []int{-1, 1, 2, 3, 3, 5}},
+	{[]int{3, 2, 1, 5, -1, 3, 0}, []int{-1, 0, 1, 2, 3, 3, 5}},
```

Comment:
Nice!
Add corner cases: empty slice, one value slice, slice with duplicated elements, sorted slice, reverse sorted slice.

Reply:
Added the corner cases.

Reply:
Just missing the duplicated elements test case. With that done, it should all be good.

Reply:
@camh-anz,its there {[]int{3, 2, 1, 5, -1, 3}, []int{-1, 1, 2, 3, 3, 5}} 3 is duplicate here.

Reply:
sorry, i missed that. all good.